lib/Ray/sources/Drawables/3D/Circle.cpp

```diff
@@ -36,4 +37,9 @@ namespace RAY::Drawables::Drawables3D
 		DrawCircle3D(this->_position, this->_radius, this->_rotationAxis,
 		             this->_rotationAngle, this->_color);
 	}
-}
\ No newline at end of file
+
+	RAY::BoundingBox Circle::getBoundingBox(void) const
+	{
+		throw RAY::Exception::NotSupportedError("Can't compute bounding box for this drawable");
```

Comment:
You should not have interface methods that are not supported. This means you break the interface's contract and this is a bad design. You could use an approximation or, if you don't need it now, throw something like a NotImplementedError.